### ISSUE TITLE
[FEAT] Gradient accumulation

### ISSUE THREAD

**Issue by pi-null-mezon** [lightly-ai/lightly-train#35]
### 💡 Is your feature request related to a problem?

Feature request.When training on GPUs with limited VRAM, users are forced to use small batch sizes. Gradient accumulation would allow users to achieve equivalent large batch sizes while working within hardware constraints.

### 🧰 Describe the solution you'd like

Add a `gradient_accumulation_steps` parameter to the `lightly_train.train()` function:  

This should:  
1. Accumulate gradients over N forward/backward passes  
2. Only perform optimizer step/weight update every N steps  
3. Automatically scale loss values if needed  

### 📝 Additional context

**Technical Implementation**:  
- Add gradient accumulation logic in optimizer step:  
```python
if (step + 1) % gradient_accumulation_steps == 0:
    optimizer.step()
    optimizer.zero_grad()
```  

**Example Usage**:  
```python
# Effective batch size = 32 (8 per step * 4 accumulation steps)
lightly_train.train(
    # ... other params ...
    batch_size=8,
    gradient_accumulation_steps=4
)
```  

**Benefits**:  
- No VRAM overhead - same memory usage as baseline batch size  
- Backwards compatible (default=1 maintains original behavior)  
- Supported by most PyTorch optimizers natively  
- Common pattern in other ML libraries (HuggingFace, PyTorch Lightning)

**Comment by guarin** [comment]
Thanks for the suggestion! We'll look into this, most pieces for it are already in place.

**Comment by pi-null-mezon** [comment]
Hi! Is there news for the topic?

**Comment by Innary** [comment]
> Thanks for the suggestion! We'll look into this, most pieces for it are already in place.

Has this feature been implemented? I really need it. If not, is there a way for me to manually insert this method?

Now I use the gradient accumulation like this:
```python
if __name__ == "__main__":
    lightly_train.train(
        trainer_args={
            "accumulate_grad_batches": 64
        },
        ...
     )
  ```

**Comment by blackpearl006** [comment]
I also really need this feature for training on GPUs with limited VRAM! 

@Innary Thanks for sharing the workaround. Just to confirm - is passing `accumulate_grad_batches` via `trainer_args` the officially supported method currently? 
```python
lightly_train.train(
    trainer_args={
        "accumulate_grad_batches": 64
    },
    batch_size=8,  # Effective batch size would be 8 * 64 = 512
    
)
```
A few questions for the maintainers:

1. Does this approach correctly scale the learning rate and other batch-size-dependent hyperparameters?
2. Are there any caveats or limitations we should be aware of when using this method? (Is this parameter overwritten anywhere internally ?)

For users who need this now, it would be helpful to document this `trainer_args `approach in the official docs until a dedicated parameter is added.
